### ISSUE TITLE
Any plan on service-worker for offline usage?

### ISSUE THREAD

**Issue by JLHwung** [vuejs/vuejs.org#771]
The [hexo-offline](https://github.com/JLHwung/hexo-offline) supports using `sw-precache` to generate a service worker for offline usage on any hexo site.

[Browser Support](https://jakearchibald.github.io/isserviceworkerready/)

**Comment by chrisvfritz** [comment]
As I just [mentioned here](https://github.com/vuejs/vuejs.org/pull/772), I'm definitely not opposed to adding offline support. 👍

**Comment by gogamwar** [comment]
I tried [hexo-offline](https://github.com/JLHwung/hexo-offline) with the following configuration:
```yaml
offline:
  maximumFileSizeToCacheInBytes: 10485760
  staticFileGlobs:
    - public/**/*.{js,html,css,png,jpg,gif,svg,eot,ttf,woff}
    - public/**/**/*.{js,html,css,png,jpg,gif,svg,eot,ttf,woff}
    - public/manifest.json
    - public/browserconfig.xml
  stripPrefix: public
```
In general, it works, but I was confused by these errors. I also was not able to properly configure caching CDN scripts and Google fonts, it works only through the browser cache.
![Network errors](https://cloud.githubusercontent.com/assets/13474134/23102247/0236665c-f6b7-11e6-86ac-2b604fcf4ea9.png)

**Comment by JLHwung** [comment]
CDN scripts should be cached using `runtimeCaching`, see https://github.com/GoogleChrome/sw-precache#runtimecaching-arrayobject

I will document the usage of `runtimeCaching` on `hexo-offline` later.

I cannot reproduce the error using the given configuration.

**Comment by gogamwar** [comment]
@JLHwung  I tried using runtimeCaching, but I could not achieve results without the use of regular expressions.
I think you can not reproduce the error since it occurs because of files that I generated for use with manifest.json (#772).
Maybe you try yourself and send a request? It would be great :flushed:

**Comment by JLHwung** [comment]
Hi morigs,

I've updated the [README](https://github.com/JLHwung/hexo-offline/blob/16cb3b85aec1eaf4d7d21c822c0456ae42526ff7/README.md) section to include CDN usage. The `runtimeCaching` works on the demo site.

I am happy to guide through, don't hesitate to contact me if there is any questions.

**Comment by chrisvfritz** [comment]
Fixed in #918
